align/cell_fabric/drc.py

```diff
@@ -94,6 +96,48 @@ def _check_via_rules(self, layer, vv):
                             if slr0[2] + space_x > slr1[0]:
                                 self.errors.append(f"Horizontal space violation on {layer}: {slr0} {slr1} {space_x}")
 
+        max_adjacent_y = self.canvas.pdk[layer].get('MaxAdjacentY', None)
+        if max_adjacent_y is not None:
+            via = getattr(self.canvas, layer)
```

Comment:
Body of the if can probably be split out into a function that takes the rect indices to query (1,3) vs. (0,2) and the max_adjacent_* value as parameters. Will reduce some code duplication but this looks functionally correct.

Reply:
Good point, there is an overlap. Yet the horizontal scanlines are bare rectangles not scanline rectangles. With the conditional checks etc, the number of line reduction is not that much.